### PR TITLE
Define schema columns to have consistency in created and loaded tables

### PR DESCRIPTION
**Pull request by sethstoudenmier**
**Description:**
Defined column order when loading data from Postgres to make sure they align with the defined schema.

**Technical details:**
Each delta_model's schema definition was moved into a dictionary to be used by both the schema and the load so that column order is guaranteed to match between the created table and the loaded data.

**Requirements for PR merge:**

1. [ ] Unit & integration tests updated
2. [ ] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
    - [ ] Frontend <OPTIONAL>
    - [ ] Operations <OPTIONAL>
    - [ ] Domain Expert <OPTIONAL>
4. [ ] Matview impact assessment completed
5. [ ] Frontend impact assessment completed
6. [ ] Data validation completed
7. [ ] Appropriate Operations ticket(s) created
8. [ ] Jira Ticket [DEV-123](https://federal-spending-transparency.atlassian.net/browse/DEV-123):
    - [ ] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/etl/management/commands/load_table_to_delta.py [hunk 1]
```diff
@@ -127,6 +127,9 @@ def handle(self, *args, **options):
                 properties=get_jdbc_connection_properties(),
             )
 
+        if table_spec.get("column_names"):
+            df = df.select(table_spec.get("column_names"))
+
```
Comment (kwhickey) [inline usaspending_api/etl/management/commands/load_table_to_delta.py:132]: Would be good to add a comment here saying why we're doing this (e.g. to indicate preserving a consistent column ordering)

**Review by kwhickey — CHANGES_REQUESTED** [review]
Looks good, just two minor suggestions
1. add a comment when ordering by cols
2. Maybe call the dictionaries `<TABLE_NAME>_COLUMNS` (or `_COLS`) rather than just `_TYPES`? I think it clarifies what the usage of that dict is moreso. e.g `AWARDS_TYPES` on its own makes me think of the different award types (contracts, loans, etc.)

File: usaspending_api/etl/management/commands/load_table_to_delta.py (hunk 1 above)
Comment (sethstoudenmier) [reply usaspending_api/etl/management/commands/load_table_to_delta.py:132]: >AWARDS_TYPES on its own makes me think of the different award types (contracts, loans, etc.

Very good point, will make that change and add comment.

**Review by kwhickey — APPROVED** [review]
Looks good

**Pull request merged**